src/main/scala/org/broadinstitute/dsde/firecloud/service/RegisterService.scala

```diff
@@ -0,0 +1,57 @@
+package org.broadinstitute.dsde.firecloud.service
+
+import akka.actor.{Actor, Props}
+import akka.pattern._
+import com.typesafe.scalalogging.slf4j.LazyLogging
+import org.broadinstitute.dsde.firecloud.Application
+import org.broadinstitute.dsde.firecloud.dataaccess.{RawlsDAO, ThurloeDAO}
+import org.broadinstitute.dsde.firecloud.model._
+import org.broadinstitute.dsde.firecloud.service.RegisterService.CreateUpdateProfile
+import org.broadinstitute.dsde.firecloud.service.PerRequest.{PerRequestMessage, RequestComplete}
+import org.broadinstitute.dsde.firecloud.utils.DateUtils
+import spray.http._
+import spray.httpx.SprayJsonSupport._
+import spray.json.DefaultJsonProtocol._
+
+import scala.concurrent.{ExecutionContext, Future}
+
+object RegisterService {
+  sealed trait ServiceMessage
+  case class CreateUpdateProfile(userInfo: UserInfo, basicProfile: BasicProfile)
```

Comment:
The name of this changed from `UpdateProfile` to `CreateUpdateProfile`. Tracing the code, I believe that the functionality has not changed and that this is now a more precise name for the action. The fact that the test expectations did not change also supports the conclusion that the functionality did not change.

I point it out here in case my conclusion is not correct or this was somehow an unintended change.